### PR TITLE
Add snapshots

### PR DESCRIPTION
**Pull request by mbfrahry**
Adding snapshots and the delete method for machine images. Will finish machine images in a later PR. 

```
make test TEST=./compute TESTARGS="-v -run=TestAccSnapshotLifeCycle"
==> Checking that code complies with gofmt requirements...
==> Checking for unchecked errors...
go test -i ./compute || exit 1
echo ./compute | \
		xargs -t -n4 go test -v -run=TestAccSnapshotLifeCycle -timeout=900s -parallel=4
go test -v -run=TestAccSnapshotLifeCycle -timeout=900s -parallel=4 ./compute
=== RUN   TestAccSnapshotLifeCycleBasic
--- PASS: TestAccSnapshotLifeCycleBasic (157.60s)
=== RUN   TestAccSnapshotLifeCycleMachineImage
--- PASS: TestAccSnapshotLifeCycleMachineImage (176.17s)
PASS
ok  	github.com/hashicorp/go-oracle-terraform/compute	333.790s
```

### CODE REVIEW

File: compute/snapshots.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	URI string `json:"uri"`
+}
+
+// CreateSnapshotInput defines an Snapshot to be created.
+type CreateSnapshotInput struct {
+	//The name of the account that contains the credentials and access details of
+	// Oracle Storage Cloud Service. The machine image file is uploaded to the Oracle
+	// Storage Cloud Service account that you specify.
+	// Optional
+	Account string `json:"account"`
```
Comment (grubernaut) [inline compute/snapshots.go]: Need `omitempty` on the optional parameter?

File: compute/snapshots.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	Quota string `json:"quota"`
+	// The state of the request.
+	State SnapshotState `json:"state"`
+	// Uniform Resource Identifier
+	URI string `json:"uri"`
+}
+
+// CreateSnapshotInput defines an Snapshot to be created.
+type CreateSnapshotInput struct {
+	//The name of the account that contains the credentials and access details of
```
Comment (grubernaut) [inline compute/snapshots.go]: space before comment string

File: compute/snapshots.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	Account string `json:"account"`
+	// Use this option when you want to preserve the custom changes you have made
+	// to an instance before deleting the instance. The only permitted value is shutdown.
+	// Snapshot of the instance is not taken immediately. It creates a machine image which
+	// preserves the changes you have made to the instance, and then the instance is deleted.
+	// Note: This option has no effect if you shutdown the instance from inside it. Any pending
+	// snapshot request on that instance goes into error state. You must delete the instance
+	// (DELETE /instance/{name}).
+	// Optional
+	Delay string `json:"delay"`
```
Comment (grubernaut) [inline compute/snapshots.go]: `omitempty` on optional parameter?

File: compute/snapshots.go [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+	// Name of the instance that you want to clone.
+	// Required
+	Instance string `json:"instance"`
+	// Specify the name of the machine image created by the snapshot request.
+	// Object names can contain only alphanumeric characters, hyphens, underscores, and periods.
+	// Object names are case-sensitive.
+	// If you don't specify a name for this object, then the name is generated automatically.
+	// Optional
+	MachineImage string `json:"machineimage"`
```
Comment (grubernaut) [inline compute/snapshots.go]: `omitempty` on optional parameter?

File: compute/snapshots.go [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+		Name: snapshotInfo.Name,
+	}
+
+	// Wait for instance to be ready and return the result
+	return c.WaitForSnapshotComplete(getInput, WaitForSnapshotCompleteTimeout)
+}
+
+// GetSnapshotInput describes the snapshot to get
+type GetSnapshotInput struct {
+	// The name of the Snapshot
```
Comment (grubernaut) [inline compute/snapshots.go]: Assuming this is required?

File: compute/snapshots.go [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+	}
+
+	deleteMachineImageRequest := &DeleteMachineImageInput{
+		Name: deleteInput.MachineImage,
+	}
+	if err := machineImagesClient.DeleteMachineImage(deleteMachineImageRequest); err != nil {
+		return fmt.Errorf("Could not delete machine image associated with snapshot: %s", err)
+	}
+
+	return nil
```
Comment (grubernaut) [inline compute/snapshots.go]: Do we need to wait for the snapshot to be deleted? Or is it a fast operation?

File: compute/snapshots_test.go [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	log.Printf("Snapshot Retrieved: %+v", snapshot)
+	if !reflect.DeepEqual(snapshot.Name, createdSnapshot.Name) {
+		t.Fatal("Snapshot Name mismatch! Got: %s Expected: %s", snapshot.Name, createdSnapshot.Name)
+	}
+}
+
```
Comment (grubernaut) [inline compute/snapshots_test.go]: Can we add one more test that covers setting both `Account` and `Delay`?

File: compute/snapshots.go (hunk 6 above)
Comment (mbfrahry) [inline compute/snapshots.go]: It's a fast operation. Only the create takes some time

File: compute/snapshots_test.go (hunk 7 above)
Comment (mbfrahry) [inline compute/snapshots_test.go]: We can try! I remember trying account but we might need some additional access. I'll give it a shot again.

File: compute/snapshots.go (hunk 5 above)
Comment (mbfrahry) [inline compute/snapshots.go]: Taken care of

**Pull request merged**